### DISCUSSION TITLE
Unable to hide fields based on FileUpload value

### DISCUSSION THREAD

**Discussion by freshleafmedia** [filamentphp/filament#5015] [Help]
I want to conditionally hide fields based on whether the `FileUpload` field has been populated or not. This doesn't appear to be working:

```php
FileUpload::make('file_name')
    ->reactive(),

TextInput::make('test')
    ->visible(fn (callable $get): bool => $get('file_name') !== []),
```

The initial state is good and the `TextInput` shows once a file is uploaded but if the file is then removed (without submitting the form) the `TextInput` remains visible.

Adding a breakpoint to the closure confirms the value of `$get('file_name')` is `[]`.

Additionally this can be taken one step further:

```php
FileUpload::make('file_name')
    ->reactive(),

TextInput::make('test')
    ->visible(function (callable $get, callable $set): bool {
        $set('test', var_export($get('file_name'), true));
        return $get('file_name') !== [];
    }),
```

Once a file is uploaded and then removed the `TextInput` shows the value `array ()` demonstrating that the value is updated dynamically.

Is this a bug or am I doing something wrong?

**Comment by danharrin** [comment]
Maybe try `->visible(fn (callable $get): bool => (bool) count(array_filter($get('file_name')))),`

**Reply by freshleafmedia** [reply]
@danharrin Thanks for the suggestion however I just tried this and it doesn't appear to work.

I have opened an issue #5019 with a full reproduction
